### PR TITLE
Implement the CRYPTO_secure_clear_free function

### PR DESCRIPTION
**Pull request by bernd-edlinger**

### CODE REVIEW

**Comment by bernd-edlinger** [comment]
the first three commits need to be back-ported to 1.1.0

File: doc/man3/OPENSSL_secure_malloc.pod [hunk 1]
```diff
@@ -76,6 +80,13 @@ It exists for consistency with OPENSSL_secure_malloc() , and
 is a macro that expands to CRYPTO_secure_free() and adds the C<__FILE__>
 and C<__LINE__> parameters..
 
+OPENSSL_secure_clear_free() is similar to OPENSSL_secure_free() except
+that it has an additional C<num> parameter which is used to clear
+the memory if it was not allocated from the secure heap.
+If CRYPTO_secure_malloc_init() is not called, this is equivalent to
+calling OPENSSL_clear_free().
+OPENSSL_secure_clear_free() was added in OpenSSL 1.1.0g.
```
Comment (richsalz) [inline doc/man3/OPENSSL_secure_malloc.pod]: Make a HISTORY section and move this line to that section.

**Comment by bernd-edlinger** [comment]
Okay, done, will go and merge this to master now.

**Comment by bernd-edlinger** [comment]
Merged to master.
Thanks!

**Pull request closed without merging**